File: frontend/src/app/features/app-statuses/models/app-status-query-parameters.ts
import { AppStatus } from './app-status-enum';
import { QueryParameters } from "../../../core/services/rest-api/paging/query-parameters";

export interface AppStatusQueryParameters extends QueryParameters {
  withAppId?: string;
  ofStatus?: AppStatus;
  recordedBefore?: string;
  recordedAfter?: string;
}

File: http-client/MCS.WatchTower.WebApi.DataTransferObjects/QueryParameters/AppStatusQueryParameters.cs
#nullable enable
using MCS.WatchTower.WebApi.DataTransferObjects.Enumerations;

namespace MCS.WatchTower.WebApi.DataTransferObjects.QueryParameters;

public record AppStatusQueryParameters(
    string? WithAppId = null,
    AppStatus? OfStatus = null,
    DateTime? RecordedBefore = null,
    DateTime? RecordedAfter = null,
    string? SearchTerm = null,
    string? OrderBy = null,
    int? Page = null,
    int? PageSize = null
) : Paging.QueryParameters(SearchTerm, OrderBy, Page, PageSize);
